src/block-data-providers/bitcoin-core/provider.ts

```diff
@@ -0,0 +1,278 @@
+            await this.setState(state);
+        }
+        this.isSyncing = false;
+    }
+
+    async getTipHeight(): Promise<number> {
+        try {
+            const body = {
+                method: 'getblockcount',
+                params: [],
+                ...this.createPartialRPCBody(),
+            };
+            return await this.request(body);
+        } catch (error) {
+            this.logger.log(`Error fetching block count`);
+            throw error;
+        }
+    }
+
+    async getBlockHash(height: number): Promise<string> {
+        try {
+            const body = {
+                method: 'getblockhash',
+                params: [height],
+                ...this.createPartialRPCBody(),
+            };
+            return await this.request(body);
+        } catch (error) {
+            this.logger.log(`Error fetching  block hash of height : ${height}`);
+            throw error;
+        }
+    }
+
+    async getBlock(hash: string, verbosity: number): Promise<Block> {
+        try {
+            const body = {
+                method: 'getblock',
+                params: [hash, verbosity],
+                ...this.createPartialRPCBody(),
+            };
+            return await this.request(body);
+        } catch (error) {
+            this.logger.log(`Error fetching block with block hash : ${hash}`);
+            throw error;
+        }
+    }
+
+    async getRawTransaction(
+        txid: string,
+        isVerbose: boolean,
+    ): Promise<BlockTransaction> {
+        try {
+            const body = {
+                method: 'getrawtransaction',
+                params: [txid, isVerbose],
+                ...this.createPartialRPCBody(),
+            };
+            return await this.request(body);
+        } catch (error) {
+            this.logger.log(
+                `Error fetching transaction with transaction id : ${txid}`,
+            );
+            throw error;
+        }
+    }
+
+    public async processBlock(height: number): Promise<Transaction[]> {
+        const parsedTransactionList: Transaction[] = [];
+        const blockHash = await this.getBlockHash(height);
+        this.logger.log(
+            `Processing block at height ${height}, hash ${blockHash}`,
+        );
+        const block = await this.getBlock(blockHash, 2);
+        for (const txn of block.tx) {
+            const parsedTransaction = await this.parseTransaction(
+                txn,
+                block.hash,
+                block.height,
+            );
+            parsedTransactionList.push(parsedTransaction);
+        }
+        return parsedTransactionList;
+    }
+
+    private async parseTransaction(
+        txn: BlockTransaction,
+        blockHash: string,
+        blockHeight: number,
+    ): Promise<Transaction> {
+        const inputs: TransactionInput[] = await Promise.all(
+            txn.vin.map(
+                async (input) => await this.parseTransactionInput(input),
+            ),
```

Comment:
pretty sure you can do something like this
```suggestion
            txn.vin.map(this.parseTransactionInput),
```

Reply:
Good catch

Reply:
We can pass `this` like this

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map